packages/gatsby-link/src/index.js

```diff
@@ -104,6 +104,13 @@ class GatsbyLink extends React.Component {
       ...rest
     } = this.props
 
+    const LOCAL_URL = /^\/(?!\/)/
+    if (process.env.NODE_ENV !== `production` && !LOCAL_URL.test(to)) {
+      console.warn(
+        `Use the Link component only for internal links. See: https://gatsby.app/internal-links`
```

Comment:
Should we add `to` prop value to warning message? It might be tedious to find instance of `Link` that uses external link without extra context. It still might not be great when `to` prop is generated dynamically, but at least you could search for it in DOM explorer to pinpoint where this is coming from

Reply:
Yeah didn't think of that! 
Not sure how to word this.  Something like: `External link ${to} was detected in a Link component. Use the Link component only for internal links. See: https://gatsby.app/internal-links`

Reply:
I am good with both 1) the comment, and 2) the proposed solution :) Thanks @oorestisime!